stdlib/public/Platform/CMakeLists.txt

```diff
@@ -98,6 +98,22 @@ add_swift_target_library(swiftGlibc ${SWIFT_STDLIB_LIBRARY_BUILD_TYPES} IS_SDK_O
     INSTALL_IN_COMPONENT sdk-overlay
     DEPENDS glibc_modulemap)
 
+add_swift_target_library(swiftMusl ${SWIFT_STDLIB_LIBRARY_BUILD_TYPES} IS_SDK_OVERLAY
+    ${swift_platform_sources}
+    POSIXError.swift
+
+    GYB_SOURCES
+      ${swift_platform_gyb_sources}
+      Musl.swift.gyb
+
+    SWIFT_COMPILE_FLAGS
+      ${SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS}
+      ${SWIFT_STANDARD_LIBRARY_SWIFT_FLAGS}
+      ${swift_platform_compile_flags}
+    LINK_FLAGS "${SWIFT_RUNTIME_SWIFT_LINK_FLAGS}"
+    TARGET_SDKS ALPINE
```

Comment:
I don't think this patch can go in like that, because `ALPINE` isn't an SDK. I also don't think `ALPINE` _should_ be an SDK. I'll talk to you offline about this part.

Reply:
Yeah, the differences here are to do with libc, but I think it's not unreasonable for Swift to refer to that as an SDK. I just think it should probably be called `MUSL` rather than `ALPINE`, since the latter is just one distribution that happens to use musl in some configuration (I guess `ANDROID` should really be `BIONIC` on that basis, as it's probably using the same libc as Fuscia?).

But perhaps more pertinently, I'd like to be in a situation where we can build a compiler that has both the `LINUX` and `MUSL` SDKs available, where right now we tend to just configure a single SDK for the UNIX variant we're building on.

Reply:
> I just think it should probably be called `MUSL` rather than `ALPINE`, since the latter is just one distribution that happens to use musl in some configuration

That's addressed in the latest commit. I'm not defining the SDK itself yet to make this PR smaller for easier review and for overall efforts to be incremental, so technically this PR is an NFC for all other platforms.

Reply:
> I'd like to be in a situation where we can build a compiler that has both the LINUX and MUSL SDKs available, where right now we tend to just configure a single SDK for the UNIX variant we're building on.

@al45tair, my suggestion in that prior comment was that we just use the `LINUX` SDK for regular glibc distros, Android, _and_ Alpine, but then provide a way to choose the libc we want to build against, similar to how you added choosing a threading library now.

I don't know what libc Fuchsia uses, but then switching the libc for other OS's should be easy, eg if one wants to use glibc on Windows someday.